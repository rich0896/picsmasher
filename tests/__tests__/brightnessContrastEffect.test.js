const { BrightnessContrastEffect } = require('../../src/effects.js');
const { createTestCanvas } = require('./helpers/testHelpers.js');

describe('BrightnessContrastEffect', () => {
    test('should apply brightness and contrast effect correctly', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new BrightnessContrastEffect({
            brightness: 20,
            contrast: 30,
        });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith(
            'Applying Brightness/Contrast Effect'
        );
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });
});
