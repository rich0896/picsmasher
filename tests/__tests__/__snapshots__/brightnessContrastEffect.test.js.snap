// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrightnessContrastEffect should apply brightness and contrast effect correctly 1`] = `ImageData { width: 200, height: 200, data: [113, 176, 239, 255, 113, 176, 239, 255, 113, 176...], length: 160000 }`;
